Tests/NIOSOCKSTests/SocksClientHandler+Tests.swift

```diff
@@ -152,5 +159,18 @@ class SocksClientHandlerTests: XCTestCase {
             XCTAssertEqual(e as? SOCKSError.ConnectionFailed, .init(reply: .serverFailure))
         }
     }
+    
+    func testDelayedConnection() {
```

Comment:
This is just the first part of `testTypicalWorkflow` with an extra assertion, i.e. we're validating the add handler and then connect flow. Should we also add a test for the reverse, i.e. connect and then add the handler?